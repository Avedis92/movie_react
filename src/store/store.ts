import { configureStore } from '@reduxjs/toolkit';
import genreReducer from './slices/genre';

export const store = configureStore({
  reducer: {
    genre: genreReducer,
  },
});

export type RootState=ReturnType<typeof store.getState>;
export type MovieDispatch=typeof store.dispatch;
