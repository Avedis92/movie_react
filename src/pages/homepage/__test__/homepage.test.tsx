import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { Provider } from 'react-redux';
import { store } from '../../../store/store';
import Homepage from '../homepage';
import '@testing-library/jest-dom/extend-expect';

const topRatedMovieList = [{
  id: 278,
  genre_ids: [18, 80],
  original_title: 'The Shawshank Redemption',
  vote_average: 8.32,
  overview: 'Framed in the 1940s for the double murder of his wife and her lover, upstanding banker Andy Dufresne begins a new life at the Shawshank prison, where he puts his accounting skills to work for an amoral warden. During his long stretch in prison, Dufresne comes to be admired by the other inmates -- including an older prisoner named Red -- for his integrity and unquenchable sense of hope.',
  poster_path: '/9O7gLzmreU0nGkIB6K3BsJbzvNv.jpg',
}];

const globalRef:any = global;
globalRef.fetch = jest.fn(() => Promise.resolve({
  json: () => Promise.resolve({
    results: topRatedMovieList,
  }),
}));
describe('Make tests on HomePage component', () => {
  test('the Navbar clicking', async () => {
    render(
      <Provider store={store}>
        <Homepage />
      </Provider>,
    );
    const topratedNavLink = screen.getByText('Top Rated');
    userEvent.click(topratedNavLink);
    const header3 = await screen.findByTestId('movieTitle');
    expect(header3).toBeInTheDocument();
    expect(header3).toHaveTextContent('The Shawshank Redemption');
  });
  test('make a snapshot test', () => {
    const {asFragment} = render(
      <Provider store={store}>
        <Homepage />
      </Provider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
