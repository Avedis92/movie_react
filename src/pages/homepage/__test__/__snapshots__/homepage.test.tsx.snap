// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Make tests on HomePage component make a snapshot test 1`] = `
<DocumentFragment>
  <div
    class="searchHeader"
  >
    <h1>
      Movies
    </h1>
    <input
      list="movieTitle"
      name="movieSearch"
      placeholder="Search"
      type="text"
    />
  </div>
  <div
    class="searchElement"
  >
    <div
      class="searchFooter"
    />
  </div>
  <div
    class="navbar"
  >
    <div
      class="movieType"
    >
      <nav>
        <div>
          <p
            class="activeNav"
            data-navtype="trendings"
          >
            Trending
          </p>
        </div>
        <div>
          <p
            class="nonActiveNav"
            data-navtype="toprated"
          >
            Top Rated
          </p>
        </div>
        <div>
          <p
            class="nonActiveNav"
            data-navtype="upcoming"
          >
            Coming Soon
          </p>
        </div>
        <div>
          <select
            class="nonActiveNav"
          >
            <option
              value="Genre"
            >
              Genre
            </option>
          </select>
        </div>
      </nav>
    </div>
    <div
      class="grids"
    >
      <button
        class="activeButton"
        type="button"
      >
        <svg
          data-testid="verticalGrid"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          style="font-size: 30px; cursor: pointer;"
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M1 2a1 1 0 0 1 1-1h2a1 1 0 0 1 1 1v2a1 1 0 0 1-1 1H2a1 1 0 0 1-1-1V2zm5 0a1 1 0 0 1 1-1h2a1 1 0 0 1 1 1v2a1 1 0 0 1-1 1H7a1 1 0 0 1-1-1V2zm5 0a1 1 0 0 1 1-1h2a1 1 0 0 1 1 1v2a1 1 0 0 1-1 1h-2a1 1 0 0 1-1-1V2zM1 7a1 1 0 0 1 1-1h2a1 1 0 0 1 1 1v2a1 1 0 0 1-1 1H2a1 1 0 0 1-1-1V7zm5 0a1 1 0 0 1 1-1h2a1 1 0 0 1 1 1v2a1 1 0 0 1-1 1H7a1 1 0 0 1-1-1V7zm5 0a1 1 0 0 1 1-1h2a1 1 0 0 1 1 1v2a1 1 0 0 1-1 1h-2a1 1 0 0 1-1-1V7zM1 12a1 1 0 0 1 1-1h2a1 1 0 0 1 1 1v2a1 1 0 0 1-1 1H2a1 1 0 0 1-1-1v-2zm5 0a1 1 0 0 1 1-1h2a1 1 0 0 1 1 1v2a1 1 0 0 1-1 1H7a1 1 0 0 1-1-1v-2zm5 0a1 1 0 0 1 1-1h2a1 1 0 0 1 1 1v2a1 1 0 0 1-1 1h-2a1 1 0 0 1-1-1v-2z"
          />
        </svg>
      </button>
      <button
        class="button"
        id="horizontal"
        type="button"
      >
        <svg
          data-testid="horizontalGrid"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          style="font-size: 30px; cursor: pointer;"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M496 288H16c-8.8 0-16 7.2-16 16v32c0 8.8 7.2 16 16 16h480c8.8 0 16-7.2 16-16v-32c0-8.8-7.2-16-16-16zm0-128H16c-8.8 0-16 7.2-16 16v32c0 8.8 7.2 16 16 16h480c8.8 0 16-7.2 16-16v-32c0-8.8-7.2-16-16-16z"
          />
        </svg>
      </button>
    </div>
  </div>
  <div
    class="movieListContainer"
  >
    <div
      class="loader"
    >
      <span />
      <span />
      <span />
    </div>
    <button
      class="seeMore"
      type="button"
    >
      See More
    </button>
  </div>
</DocumentFragment>
`;
